src/Limiters/quasimonotone.jl

```diff
@@ -0,0 +1,348 @@
+)
+    q_bounds_nbr = limiter.q_bounds_nbr
+    (_, _, _, Nv, Nh) = size(q_bounds_nbr)
+    if limiter.ghost_buffer isa Spaces.GhostBuffer
+        q_bounds_ghost = limiter.ghost_buffer.recv_data
+
+        for h in 1:Nh
+            for v in 1:Nv
+                slab_q_bounds = slab(q_bounds_nbr, v, h)
+                q_min = slab_q_bounds[1]
+                q_max = slab_q_bounds[2]
+                for gidx in Topologies.ghost_neighboring_elements(topology, h)
+                    ghost_slab_q_bounds = slab(q_bounds_ghost, v, gidx)
+                    q_min = rmin(q_min, ghost_slab_q_bounds[1])
+                    q_max = rmax(q_max, ghost_slab_q_bounds[2])
+                end
+                slab_q_bounds_nbr = slab(q_bounds_nbr, v, h)
+                slab_q_bounds_nbr[1] = q_min
+                slab_q_bounds_nbr[2] = q_max
+            end
+        end
+    end
+    return nothing
+end
+
+"""
+    compute_bounds!(limiter::QuasiMonotoneLimiter, ρq::Field, ρ::Field)
+
+Compute the tracer bounds for the tracer concentration `q` based on the fields `ρq` and `ρ`.
+"""
+function compute_bounds!(
+    limiter::QuasiMonotoneLimiter,
+    ρq::Fields.Field,
+    ρ::Fields.Field,
+)
+    topology = Spaces.topology(axes(ρq))
+    compute_bounds!(
+        limiter,
+        Fields.field_values(ρq),
+        Fields.field_values(ρ),
+        topology,
+    )
+end
+function compute_bounds!(
+    limiter::QuasiMonotoneLimiter,
+    ρq,
+    ρ,
+    topology::Topologies.AbstractTopology,
+)
+    compute_element_bounds(limiter, ρq, ρ)
+    if limiter.ghost_buffer isa Spaces.GhostBuffer
+        Spaces.fill_send_buffer!(
+            topology,
+            limiter.q_bounds,
+            limiter.ghost_buffer,
+        )
+        ClimaComms.start(limiter.ghost_buffer.graph_context)
+    end
+    compute_neighbor_bounds_local!(limiter, topology)
+    if limiter.ghost_buffer isa Spaces.GhostBuffer
+        ClimaComms.finish(ghost_buffer.graph_context)
+        compute_neighbor_bounds_ghost!(limiter, topology)
+    end
+end
+
+
+
+
+"""
+    apply_limiter!(ρq, ρ, limiter::QuasiMonotoneLimiter)
+
+Apply the computed bounds of the tracer concentration (`q`) in the limiter to `ρq`.
+"""
+function apply_limiter!(
+    ρq::Fields.Field,
+    ρ::Fields.Field,
+    limiter::QuasiMonotoneLimiter,
+)
+
+    space = axes(ρq)
+
+    # Initialize temp variables
+
+    ρq_data = Fields.field_values(ρq)
+    ρ_data = Fields.field_values(ρ)
+    WJ_data = Spaces.local_geometry_data(space).WJ
+
+    (_, _, _, Nv, Nh) = size(ρq_data)
+    for h in 1:Nh
+        for v in 1:Nv
+            slab_ρ = slab(ρ_data, v, h)
+            slab_ρq = slab(ρq_data, v, h)
+            slab_wj = slab(WJ_data, v, h)
+            slab_q_bounds = slab(limiter.q_bounds_nbr, v, h)
+            apply_limit_slab!(slab_ρq, slab_ρ, slab_wj, slab_q_bounds)
+        end # end of horz elem loop
+    end # end of vert level loop
+    return ρq
+end
+
+function apply_limit_slab!(
```

Comment:
Please add documentation to this function. I see that here is where the bulk of the original implementation was. 
By any chance, did you use `git mv` when renaming this file? It helps track the history of the changes. Especially on GitHub, which is very finicky about showing history of files after rebases already (but I use a Chrome extension to see those). So please consider using `git mv` if you haven't done so. Thank you!

Reply:
I did, but `git mv` doesn't actually do anything fancy internally: because git is content-addressed, it is the same as deleting and adding a new file: https://git.wiki.kernel.org/index.php/GitFaq#Why_does_Git_not_.22track.22_renames.3F

Reply:
Yes, I know. I resorted using this Chrome extension that sometimes helps 
https://chrome.google.com/webstore/detail/git-history-browser-exten/laghnmifffncfonaoffcndocllegejnf?hl=en